### ISSUE TITLE
[TF2] UnicodeDecodeError when using tf.saved_model.save

### ISSUE THREAD

**Issue by csachs** [tensorflow/tensorflow#31272]
**System information**
- Have I written custom code: Yes.
- OS Platform and Distribution: Ubuntu 18.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version: 2.0.0-dev20190731
- Python version: 3.6.7
- CUDA/cuDNN version: 10.1
- GPU model and memory: GTX 1080 Ti

**Describe the current behavior**
Code crashes with following exception:
```
Traceback (most recent call last):
  File "mwe.py", line 10, in <module>
    tf.saved_model.save(model, '/tmp/model')
  File "***/lib/python3.6/site-packages/tensorflow_core/python/saved_model/save.py", line 869, in save
    saveable_view, asset_info.asset_index)
  File "***/lib/python3.6/site-packages/tensorflow_core/python/saved_model/save.py", line 624, in _serialize_object_graph
    _write_object_proto(obj, obj_proto, asset_file_def_index)
  File "***/lib/python3.6/site-packages/tensorflow_core/python/saved_model/save.py", line 663, in _write_object_proto
    metadata=obj._tracking_metadata)
  File "***/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/training.py", line 2816, in _tracking_metadata
    metadata = json.loads(super(Model, self)._tracking_metadata)
  File "***/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 2198, in _tracking_metadata
    metadata['config'] = self.get_config()
  File "***/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/network.py", line 878, in get_config
    layer_config = layer.get_config()
  File "***/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 2378, in get_config
    'node_def': self.node_def.SerializeToString().decode('utf-8'),
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 68: invalid start byte
```

This seems to be a regression; I was running code I had working ago some weeks/months with earlier TF2 nightlies, and this appears to me as unexpected behavior.

Might be introduced by https://github.com/tensorflow/tensorflow/commit/7cc180f107f142432358ac33787466de90afd776

To me, `<protobuf object>` `.SerializeToString().decode('utf-8')` seems incorrect, since `SerializeToString()` can return arbitrary binary data (https://developers.google.com/protocol-buffers/docs/pythontutorial#parsing-and-serialization "Note that the bytes are binary, not text; we only use the str type as a convenient container."), decodability into UTF-8 cannot be guaranteed. This bug might have gone unnoticed, as many other serializations just happen to be be UTF-8 decodable …

**Describe the expected behavior**

A properly saved model.

**Code to reproduce the issue**
```python
import tensorflow as tf
import tensorflow.keras as keras

input_ = keras.layers.Input((None, None, None), dtype='float32')

output = tf.image.extract_patches(input_, sizes=(1,128,128,1), strides=(1,128,128,1), rates=(1,1,1,1), padding='VALID')

model = keras.Model(inputs=[input_], outputs=[output])

tf.saved_model.save(model, '/tmp/model')
```

**Comment by ymodak** [comment]
I was able to reproduce the error with TF 2.0, beta1 and nightly.

**Comment by csachs** [comment]
This seems to have been fixed in 39bc7bcf94983261a3ee8a72802f5de056728a9c .
